### PR TITLE
[tinyfiledialogs] Fix tinyfiledialogs not being fetchable from sourceforge

### PR DESCRIPTION
**Pull request by MaxKruse**
**Describe the pull request**
Fix tinyfiledialogs not being fetchable from sourceforge. Use github instead.

- What does your PR fix? Fixes issue #
No issue was created, but i encountered issues on my local machine.

- Which triplets are supported/not supported? Have you updated the CI baseline?
No update outside of git url

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

**Comment by PhoebeHui** [comment]
The source seems move to https://sourceforge.net/p/tinyfiledialogs/code/ci/master/tree/

File: ports/tinyfiledialogs/portfile.cmake [hunk 1]
```diff
@@ -4,8 +4,8 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
 vcpkg_from_git(
     OUT_SOURCE_PATH SOURCE_PATH
-    URL https://git.code.sf.net/p/tinyfiledialogs/code
-    REF 331f93e903e0555399bf81479114dda978a77d7c
+    URL https://github.com/native-toolkit/tinyfiledialogs
+    REF f2379486f0e516cab528fc49991ad1b36c2e2831
 )
```
Comment (NancyLi1013) [inline ports/tinyfiledialogs/portfile.cmake]: Could you please update the source as @PhoebeHui suggested in the the comment?
Also we should update the function as` vcpkg_download_distfile()` and add `vcpkg_extract_source_archive_ex()`.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @MaxKruse thanks for this PR.
Since there is no open issue to follow this problem.
Could you please open an issue about this?

**Comment by MaxKruse** [comment]
> Hi @MaxKruse thanks for this PR.
> Since there is no open issue to follow this problem.
> Could you please open an issue about this?
> 
> What's more, please also update the version and add homepage in [CONTROL ](https://github.com/microsoft/vcpkg/blob/master/ports/tinyfiledialogs/CONTROL)file.

#9665 is the issue number. Hope i didnt mess up anything.

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for the PR!

**Comment by vicroms** [comment]
We need to find a more permanent solution to acquire this library's sources, the REFs expire too often.

Maybe contacting the maintainer and telling them about the advantages of providing versioned archives.

**Pull request merged**